			     BASH PATCH REPORT
			     =================

Bash-Release:	5.2
Patch-ID:	bash52-031

Bug-Reported-by:	Grisha Levit <grishalevit@gmail.com>
Bug-Reference-ID:	<CAMu=BrrD52xxF7C9Ke1fosAurHq1kz82t7FcHsrPBw9CbAst6A@mail.gmail.com>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2023-06/msg00045.html

Bug-Description:

There is a memory leak in the code that implements the optimized $(<file)
expansion for some code paths.

Patch (apply with `patch -p0'):

*** ../bash-5.2-patched/builtins/evalstring.c	Tue Dec 13 12:53:21 2022
--- builtins/evalstring.c	Tue Nov 28 17:25:39 2023
***************
*** 763,766 ****
--- 773,779 ----
    if (fnp)
      *fnp = fn;
+   else
+     free (fn);
+ 
    return fd;
  }

*** ../bash-5.2/patchlevel.h	2020-06-22 14:51:03.000000000 -0400
--- patchlevel.h	2020-10-01 11:01:28.000000000 -0400
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 30
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 31
  
  #endif /* _PATCHLEVEL_H_ */

